### DISCUSSION TITLE
Ability to configure injections based of a key-value pair from a  configuration file

### DISCUSSION THREAD

**Discussion by kakaday22** [allrod5/injectable#110] [Ideas]
This library can greatly improve if we could configure our injections based of a profile similar to Java's SpringBoot. 

during the injection we provide the key and internally we look for the value (qualifier name) inside the active profile

## Example
set environment variable `INJECTABLE_PROFILE = dev`

### Configuration Example
```yaml
default: dev
profiles:
  dev:
    database: MssqlDB
  prod:
    database: SqlDB
```

### Code
internally the library  looks for the configuration file and finds the  configuration for the active profile

* mssql.py
```python
@injectable(qualifier="MssqlDB")
class MssqlDB:
   ... implementation
```
* sql.py
```python
@injectable(qualifier="MssqlDB")
class SqlDB:
   ... implementation
```

```python
database = inject(key="database")
```

since profile is dev it injects  MsqqlDB

Thoughts?

**Comment by roo-oliv** [comment]
Hi @kakaday22, thank you for taking the time to contribute ❤️ 

Indeed, this suggestion seems to fit well with the framework! I'll dedicate some time in the following days to explore an implementation of this!

If you wish, feel free to submit a PR or implementation suggestions as well and I'll review them. Nevertheless loved the suggestion!
